benches/poseidon_benches.rs

```diff
@@ -13,39 +13,87 @@ use merlin::Transcript;
 use hades252::scalar;
 
-/// This function allows us to bench the whole process of
-/// digesting a vec of `Scalar`. We put together the process
-/// of creating the scalar array + digesting the info to be able
-/// to compare the results against the unoptimized version.
-#[inline]
-fn digest_one() -> () {
-    let s = Scalar::random(&mut thread_rng());
-    scalar::hash(&[s]).unwrap();
-}
 
-#[inline]
-fn digest(n: usize) -> () {
-    let s: Vec<Scalar> = (0..n).map(|_| Scalar::random(&mut thread_rng())).collect();
-    scalar::hash(&s).unwrap();
-}
+mod scalar_benches {
+    use super::*;
 
-pub fn hash_one_scalar(c: &mut Criterion) {
-    c.bench_function("Hashing single `Scalar`", |b| b.iter(|| digest_one()));
-}
+    /// This function allows us to bench the whole process of
+    /// digesting a vec of `Scalar`. We put together the process
+    /// of creating the scalar array + digesting the info to be able
+    /// to compare the results against the unoptimized version.
+    #[inline]
+    fn digest_one() -> () {
+        let s = Scalar::random(&mut thread_rng());
+        scalar::hash(&[s]).unwrap();
+    }
+
+    #[inline]
+    fn digest(n: usize) -> () {
+        let s: Vec<Scalar> = (0..n).map(|_| Scalar::random(&mut thread_rng())).collect();
+        scalar::hash(&s).unwrap();
+    }
 
-pub fn hash_two_scalars(c: &mut Criterion) {
-    c.bench_function("Hashing two `Scalar`", |b| b.iter(|| digest(black_box(2))));
+    pub fn hash_one_scalar(c: &mut Criterion) {
+        c.bench_function("Hashing single `Scalar`", |b| b.iter(|| digest_one()));
+    }
+
+    pub fn hash_two_scalars(c: &mut Criterion) {
+        c.bench_function("Hashing two `Scalar`", |b| b.iter(|| digest(black_box(2))));
+    }
+
+    pub fn hash_four_scalars(c: &mut Criterion) {
+        c.bench_function("Hashing four `Scalar`", |b| b.iter(|| digest(black_box(4))));
+    }
 }
 
-pub fn hash_four_scalars(c: &mut Criterion) {
-    c.bench_function("Hashing four `Scalar`", |b| b.iter(|| digest(black_box(4))));
+mod lc_benches {
+    use super::*;
+
+    /// This function allows us to bench the whole process of
+    /// digesting a vec of `LinearCombination`. We put together the process
+    /// of creating the linearcombination array + digesting the info to be able
+    /// to compare the results against the unoptimized version.
+    #[inline]
+    fn digest_one() -> () {
+        let s = LinearCombination::from(Scalar::random(&mut thread_rng()));
+
+        let mut verifier_transcript = Transcript::new(b"");
+        let mut verifier = Verifier::new(&mut verifier_transcript);
+
+        linear_combination::hash(&mut verifier, &[s]).unwrap();
+    }
+
+    #[inline]
+    fn digest(n: usize) -> () {
+        let s: Vec<LinearCombination> = (0..n).map(|_| LinearCombination::from(Scalar::random(&mut thread_rng()))).collect();
+        
+        let mut verifier_transcript = Transcript::new(b"");
+        let mut verifier = Verifier::new(&mut verifier_transcript);
```

Comment:
This block that creates the `Verifier` makes the benchmark take a lot of time to run and obfuscates the time gains a little bit.

Reply:
As we discussed, it's not the block that creates the `Verifier` that impact, however we can always remove that from the benchmark as mentioned in PR #26